packages/@aws-cdk/aws-rds/lib/proxy.ts

```diff
@@ -408,20 +408,30 @@ export class DatabaseProxy extends cdk.Resource
     this.endpoint = this.resource.attrEndpoint;
 
     let dbInstanceIdentifiers: string[] | undefined;
-    if (bindResult.dbClusters) {
-      // support for only instances of a single cluster
-      dbInstanceIdentifiers = bindResult.dbClusters[0].instanceIdentifiers;
-    } else if (bindResult.dbInstances) {
+    if (bindResult.dbInstances) {
       // support for only single instance
       dbInstanceIdentifiers = [ bindResult.dbInstances[0].instanceIdentifier ];
     }
 
-    new CfnDBProxyTargetGroup(this, 'ProxyTargetGroup', {
+    let dbClusterIdentifiers: string[] | undefined;
+    if (bindResult.dbClusters) {
+      dbClusterIdentifiers = bindResult.dbClusters.map((c) => c.clusterIdentifier);
+    }
+
+    if (!!dbInstanceIdentifiers && !!dbClusterIdentifiers) {
+      throw new Error('Cannot specify both dbInstanceIdentifiers and dbClusterIdentifiers');
+    }
+
+    const proxyTargetGroup = new CfnDBProxyTargetGroup(this, 'ProxyTargetGroup', {
       dbProxyName: this.dbProxyName,
       dbInstanceIdentifiers,
-      dbClusterIdentifiers: bindResult.dbClusters?.map((c) => c.clusterIdentifier),
+      dbClusterIdentifiers,
       connectionPoolConfigurationInfo: toConnectionPoolConfigurationInfo(props),
     });
+
+    // Currently(2020-07-04), this property must be set to default.
+    // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbproxytargetgroup.html#TargetGroupName-fn::getatt
+    proxyTargetGroup.addOverride('Properties.TargetGroupName', 'default');
```

Comment:
Overrides are set for CloudFormation properties that are not yet supported in the CDK. This is a CloudFormation return value. 

Why is this being set?

Reply:
There seems to be a difference between [the CloudFormation resource document](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbproxytargetgroup.html#TargetGroupName-fn::getatt) and what CloudFormation actually does.

Currently CloudFormation requires `TargetGroupName` in props of `ProxyTargetGroup`, but it is not in `CfnProxyTargetGroupProps`

There is no normal way to set `TargetGroupName`, but through `addOverride`.

![image](https://user-images.githubusercontent.com/15084045/86578189-605ecc80-bfb6-11ea-9c1e-f6add9e6168d.png)

The behavior seems to be changed around DB Proxy's GA.

Reply:
Thanks for the explanation.